src/Sarif/Autogenerated/Run.cs

```diff
@@ -151,6 +154,7 @@ public SarifNodeKind SarifNodeKind
         /// </summary>
         public Run()
         {
+            this.ColumnKind = ColumnKind.UnicodeCodePoints;
```

Comment:
>this.ColumnKind = ColumnKind.UnicodeCodePoints [](start = 12, length = 46)

ok, this is an important apparent JSON.NET gap. in JSON.NET, you can provide an argument in a JSON convert settings object that specifies that default values should *populate* if missing on deserialization. the problem with that is that you need to explicitly create and pass the appropriate settings object (and we've attempted to make everything work with a single API call with no parameterization of this kind). 

to work around the problem, we provide a programmatic equivalent of populate by setting the default value in the ctor. when JSON.NET itself is deserialization, it will overwrite this property in cases where the JSON has an explicit value.  a fix to jschema #57 should provide this fix as well (and not just the default value attribute that elides property values from persisted JSON if they map to the default).

Reply:
I assume you are talking about [JsonSerializerSettings.DefaultValueHandling](https://www.newtonsoft.com/json/help/html/SerializationSettings.htm#DefaultValueHandling), whose `Include` value means that "The Json.NET deserializer will continue setting a field/property if the JSON value is the same as the default value."

But it goes on to say "DefaultValueHandling can also be customized on individual properties with JsonPropertyAttribute." So I think we can fix this by emitting
```
[JsonProperty DefaultValueHandling = DefaultValueHandling.Include)]
```
on every property.

---
In reply to: [239879240](https://github.com/Microsoft/sarif-sdk/pull/1160#discussion_r239879240) [](ancestors = 239879240)

Reply:
I added that to [https://github.com/Microsoft/jchema/issues/57](https://github.com/Microsoft/jchema/issues/57).

---
In reply to: [239910615](https://github.com/Microsoft/sarif-sdk/pull/1160#discussion_r239910615) [](ancestors = 239910615,239879240)